src/paho/mqtt/client.py

```diff
@@ -993,10 +1019,7 @@ def loop(self, timeout=1.0, max_packets=1):
             # Can occur if we just reconnected but rlist/wlist contain a -1 for
             # some reason.
             return MQTT_ERR_CONN_LOST
-        except KeyboardInterrupt:
-            # Allow ^C to interrupt
-            raise
-        except:
+        except Exception:  # Still allows ^C (KeyboardInterrupt) to interrupt
```

Comment:
I would clarify this as "Note that KeyboardInterrupt, etc. are not derived from Exception`